backend/experiment/actions/form.py

```diff
@@ -38,91 +62,280 @@ def __init__(
         self.submits = submits
         self.style = style
 
-    def action(self):
+    def action(self) -> Dict[str, Any]:
         if settings.TESTING and self.result_id:
             from result.models import Result
+
             result = Result.objects.get(pk=self.result_id)
             if result and result.expected_response:
                 self.expected_response = result.expected_response
         return self.__dict__
 
 
 class NumberQuestion(Question):
-    def __init__(self, input_type='number', min_value=0, max_value=120, **kwargs):
+    """A question that accepts numeric input.
+
+    Args:
+        input_type (str): Type of number input (default: 'number')
+        min_value (int): Minimum allowed value
+        max_value (int): Maximum allowed value
+        **kwargs: Additional Question arguments
+    """
+
+    def __init__(self, input_type: str = "number", min_value: int = 0, max_value: int = 120, **kwargs) -> None:
         super().__init__(**kwargs)
         self.min_value = min_value
         self.max_value = max_value
         self.input_type = input_type
-        self.view = 'STRING'
+        self.view = "STRING"
 
 
 class TextQuestion(Question):
-    def __init__(self, input_type='text', max_length=64, **kwargs):
+    """A question that accepts text input.
+
+    Args:
+        input_type (str): Type of text input (default: 'text')
+        max_length (int): Maximum character length
+        **kwargs: Additional Question arguments
+    """
+
+    def __init__(self, input_type: str = "text", max_length: int = 64, **kwargs) -> None:
         super().__init__(**kwargs)
-        self.max_length = max_length # the maximum length of the question's answer in characters
+        self.max_length = max_length  # the maximum length of the question's answer in characters
         self.input_type = input_type
-        self.view = 'STRING'
+        self.view = "STRING"
 
 
 class BooleanQuestion(Question):
-    def __init__(self, choices=None, **kwargs):
+    """A yes/no question component.
+
+    Args:
+        choices (Optional[Dict[str, str]]): Custom yes/no labels
+        **kwargs: Additional Question arguments
+
+    Example:
+        ```python
+        question = BooleanQuestion(
+            key="is_student",
+            question="Are you a student?",
+            choices={
+                "no": "No", # or _("No") for translation
```

Comment:
Perhaps mention that `_("No")` is the default? And perhaps use "Nope" and "Yep" as examples, to make sure that users won't think it's necessary to define choices if they are happy with (translatable) yes/no.